src/collect_columns/collect_columns.py

```diff
@@ -46,24 +47,20 @@ def collect_columns(count_tables: List[Path], feature_column: int,
     # If no sample names are given use table basenames instead.
     if names is None:
         names = [path.name for path in count_tables]
-    # Convert tables_have_headers to conform with the pandas parameter.
-    header = 0 if tables_have_headers else None
-    # Load in the first table.
-    merged_table = pd.read_csv(count_tables.pop(0), sep=sep, header=header,
-                               index_col=feature_column,
-                               usecols=[feature_column, value_column])
-    merged_table.columns = [names.pop(0)]
-    merged_table.index.names = ["feature"]
-    # Merge the other tables into the first.
+    counts = {}
     for i, table in enumerate(count_tables):
-        sample_table = pd.read_csv(table, sep=sep, header=header,
-                                   index_col=feature_column,
-                                   usecols=[feature_column, value_column])
-        sample_table.columns = [names[i]]
-        sample_table.index.names = ["feature"]
-        merged_table = merged_table.merge(sample_table, left_index=True,
-                                          right_index=True, how="outer")
-    return merged_table
+        reader = csv.reader(table.open(), delimiter=sep)
+        if tables_have_headers is True:
+            next(reader)
+        counts[names[i]] = {}
```

Comment:
For more clarity you could use `column_name = names[i]` and then use `column_name` instead of `names[i]` in the rest of the code, which is more easily understandable and avoids some brace nesting confusion. `counts[names[i]][feature]` -> `counts[column_name][feature]`